File: Frontend/src/app/models/usuario.ts
import { Rol } from "./rol";
import { Pregunta } from "./pregunta";

export class Usuario {

    public id: number;
    public createdAt: Date;
    public nombre: string;
    public password: string;
    public correo: string;

    public rol: Rol;
    public preguntas: Pregunta[];

    public constructor(json?: any) {

        Object.assign(this, json);

        if (this.createdAt) { this.createdAt = new Date(this.createdAt) };
        if (this.rol) { this.rol = new Rol(this.rol) };
        if (this.preguntas) { this.preguntas = this.preguntas.map(pregunta => new Pregunta(pregunta)) };

    }

    public getJSON(): any {
        return {
            id: this.id,
            createdAt: this.createdAt,
            nombre: this.nombre,
            password: this.password,
            correo: this.correo
        };
    }

}

File: Backend/src/main/java/aqct/backend/model/Usuario.java
package aqct.backend.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import java.io.Serializable;
import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import org.hibernate.annotations.CreationTimestamp;

@Entity
@Table(name = "usuario")
public class Usuario implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(name = "nombre", unique = true)
    private String nombre;

    @Column(name = "correo", unique = true)
    private String correo;

    @Column(name = "password")
    private String password;

    @Column(name = "created_at")
    @CreationTimestamp
    private Timestamp createdAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "rol_id")
    private Rol rol;

    @OneToMany
    @JoinColumn(name = "usuario_id")
    @JsonManagedReference
    private List<Pregunta> preguntas = new ArrayList<>();

    public Usuario() {

    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getCorreo() {
        return correo;
    }

    public void setCorreo(String correo) {
        this.correo = correo;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public Timestamp getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Timestamp createdAt) {
        this.createdAt = createdAt;
    }

    public Rol getRol() {
        return rol;
    }

    public void setRol(Rol rol) {
        this.rol = rol;
    }

}
